htlcswitch/switch.go

```diff
@@ -2106,6 +2245,35 @@ func (s *Switch) addLiveLink(link ChannelLink) {
 		s.interfaceIndex[peerPub] = make(map[lnwire.ChannelID]ChannelLink)
 	}
 	s.interfaceIndex[peerPub][link.ChanID()] = link
+
+	if IsAlias(link.ShortChanID()) {
```

Comment:
It's worth pointing out that when the zero-conf link is created, we'll need to assign the link some "base" `ShortChannelID` to use. I used the alias here. The maps in the switch are currently 1:1 but it's easy to change that to a N:1 and 1:N